worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala

```diff
@@ -971,6 +986,14 @@ private[celeborn] class Worker(
     }
     serverBootstraps
   }
+
+  private def isDecommissioning: Int = {
+    if (shutdown.get() && workerStatusManager.currentWorkerStatus.getState == State.InDecommission) {
```

Comment:
State `InDecommissionThenIdle` should also be considered as `isDecommissioning`.